pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -354,6 +390,24 @@ func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, c
 	return dataFromTemplate, nil
 }
 
+func convertRegistryMapToSliceInOrder(registryMap map[int]map[string]interface{}) []map[string]interface{} {
+	registries := make([]map[string]interface{}, len(registryMap))
+	for k, v := range registryMap {
+		registries[k] = v
+	}
+	return registries
+}
+
+func getIndexFromQuestion(question string) (int, error) {
+	var index int
+	err := fmt.Errorf("cannot get index from the question: %s", question)
+	target := regexp.MustCompile(`\[\d+\]`).FindString(question)
```

Comment:
I switched from `regexp.Compile`  to ` regexp.MustCompile` because the regular expression is hardcoded so it cannot fail to be complied.

Reply:
@thedadams , this was changed based on something I commented on before.
https://github.com/rancher/rancher/pull/36084#discussion_r782439144

I'm fine switching back to `regexp.Compile` provided we handle any errors it may return. I generally find using `MustCompile` safe when hardcoded regexps are used based on the documentation note from https://pkg.go.dev/regexp#MustCompile: _MustCompile is like Compile but panics if the expression cannot be parsed._

Reply:
switched back to `regex.Complie` with the proper handle of errors.